### PR TITLE
button refactor

### PR DESCRIPTION
**Pull request by tballmsft**

### CODE REVIEW

File: services/button.c [hunk 1]
```diff
@@ -58,9 +59,6 @@ void btn_process(srv_t *state) {
 }
 
 void btn_handle_packet(srv_t *state, jd_packet_t *pkt) {
-    if (sensor_handle_packet_simple(state, pkt, &state->pressed, sizeof(state->pressed)))
```
Comment (mmoskal) [inline services/button.c]: you still need that - this handles the case when the client asks for streaming

File: services/button.c [hunk 2]
```diff
@@ -58,9 +59,6 @@ void btn_process(srv_t *state) {
...
-        return;
-
     switch(service_handle_register(state, pkt, button_regs)) {
```
Comment (mmoskal) [inline services/button.c:62]: could you "format document" in vscode?

File: services/button.c [hunk 3]
```diff
@@ -25,20 +33,19 @@ static void update(srv_t *state) {
         pin_set(state->backlight_pin, state->pressed);
         if (state->pressed) {
             jd_send_event(state, JD_BUTTON_EV_DOWN);
-            state->press_time = now;
+            state->press_time = now / 1000;
```
Comment (mmoskal) [inline services/button.c:36]: the internal timestamps should be in `us`; this way overflows won't affect you (unless someone holds the button for over an hour); after subtraction of timestamps you can convert to `ms`

File: services/button.c (hunk 1 above)
Comment (tballmsft) [reply services/button.c]: oh, my bad. thanks.

File: services/button.c (hunk 2 above)
Comment (tballmsft) [reply services/button.c:62]: yep

File: services/button.c (hunk 3 above)
Comment (tballmsft) [reply services/button.c:36]: got it.

**Comment by tballmsft** [comment]
doh!

**Pull request merged**